app/containers/ExplorePage.tsx

```diff
@@ -51,45 +52,110 @@ function chooseFile(
 export default function ExplorePage() {
   const { t } = useTranslation();
   const [filePath, setFilePath] = useState<string>();
+  const [filters, setFilters] = useState<object>({
+    activeAnimals: [],
+    activeCameras: [],
+    activeStations: [],
+    activeChecks: []
+  });
   const [data, setData] = useState<undefined | ObservationsData>();
+  const handleFilters = val => {
+    setFilters({ ...filters, ...val });
+  };
+
+  const filterCondition = (needle, haystack) => {
+    if (haystack.length === 0) return true;
+    return haystack.map(entry => entry.value).includes(needle);
+  };
+
+  const getFilteredData = options => {
+    let filtered = options.observations;
+
+    if (typeof filters !== 'undefined') {
+      filtered = filtered.filter(entry => {
+        const validEntry =
```

Comment:
i think that inlining this variable would actually increase readability;
you could also remove `{ }` then